### PR TITLE
Add support for multiple network configuration options

### PR DESCRIPTION
**Pull request by ryan-dyer-sp**
### What does this PR do?
Updates the network integration to support the current set of options available.

### Motivation
need support for ENA metrics in the network integration via puppet

### Describe your test plan
Just manual testing using the module and updating various variables to ensure they rendered conf.yaml contained the correct data.

### CODE REVIEW

File: manifests/integrations/network.pp [hunk 1]
```diff
@@ -20,10 +20,25 @@
...
-  String $excluded_interface_re = '',
-  Boolean $combine_connection_states = true,
+  Boolean $collect_connection_state          = false,
+  Boolean $collect_connection_queues         = false,
+  Array[String] $excluded_interfaces         = ['lo','lo0'],
+  String $excluded_interface_re              = '',
+  Boolean $combine_connection_states         = true,
+  Boolean $collect_rate_metrics              = true,
+  Boolean $collect_count_metrics             = false,
+  String $conntrack_path                     = '/usr/sbin/conntrack',
```
Comment (bkabrda) [inline manifests/integrations/network.pp:30]: The default for `conntrack_path` should be null.

File: manifests/integrations/network.pp [hunk 2]
```diff
@@ -20,10 +20,25 @@
...
+  Boolean $use_sudo_conntrack                = true,
+  Array[String] $whitelist_conntrack_metrics = [],
+  Array[String] $blacklist_conntrack_metrics = [],
+  Boolean $collect_aws_ena_metrics           = false,
+  Boolean $collect_ethtool_metrics           = false,
```
Comment (bkabrda) [inline manifests/integrations/network.pp:35]: Hmm, I don't see this configuration value anywhere in the sample config or the code of the integration. Where did you find it?

File: templates/agent-conf.d/network.yaml.erb [hunk 3]
```diff
@@ -14,3 +15,47 @@ instances:
...
+    collect_ethtool_metrics: <%= @collect_ethtool_metrics %>
+<% if !@tags.empty? -%>
+    tags:
+<%- (@tags.each do |tag| -%>
+      - <%= tag %>
+<% end ) -%>
+<% end -%>
+<% if !@service.empty? -%>
+    service: <%= @service %>
+<% end -%>
```
Comment (bkabrda) [inline templates/agent-conf.d/network.yaml.erb:44]: `service` key should live under `init_config`, not under `instances`.

**Review by bkabrda — CHANGES_REQUESTED** [review]
Hi 👋. Thanks a lot for submitting this PR. Overall it looks good, but I requested some minor changes (see the comments inline) to make it consistent with the actual defaults. Once these get fixed, I can merge.

File: templates/agent-conf.d/network.yaml.erb (hunk 3 above)
Comment (ryan-dyer-sp) [reply templates/agent-conf.d/network.yaml.erb:44]: https://github.com/DataDog/integrations-core/blob/ba25dd5cfc9f6769435d60bdd2b92d7c5f45bee1/network/datadog_checks/network/data/conf.yaml.default#L135

Comment (bkabrda) [reply templates/agent-conf.d/network.yaml.erb:43]: You're right, please disregard this part.

**Comment by bkabrda** [comment]
@ryan-dyer-sp thanks for doing the changes. Could you possibly rebase on top of latest `main` branch? This should fix the CI - once that is green, I can merge the PR. Thanks!

**Comment by ryan-dyer-sp** [comment]
@bkabrda rebased and CI now passing

**Review by bkabrda — APPROVED** [review]
LGTM now. Thanks for the PR!

**Pull request merged**